### ISSUE TITLE
Idea: expose recently used variables to the DSL

### ISSUE THREAD

**Issue by andymeneely** [andymeneely/squib#89]
It would be interesting to have "global" (i.e. deck) variables that are set after most method calls. Something like `last_x`, `last_y`, `last_width`, `last_height`

Now of course you can/should do this in your `layout.yml` with extends. But there are some cases that are dynamic in nature (e.g. the extents that get returned from `text`). This would, say, allow you to set the subtitle to be 0.125 inches below the title, and it would change properly when you change the title font.

Here's an illustrative example:

```ruby
text str: 'Title', font: 'Sans 64'  # who knows where this ends??
text str: 'subtitle', y: last_y + last_height + 50 #50 pixels below where the subtitle ends
```

This helps with rapid prototyping too. Don't have to open up the image file and start looking pixels, or guess-and-checking.

**Comment by andymeneely** [comment]
I'd like to play around with this. One issue here is singleton expansion. The above example makes sense if we're talking about the same thing on the same cards, but in reality it'll differ from card to card. So instead it'll have to be something like:

```ruby
def plus(arr1, arr2)
  arr1.zip(arr2).map { |x| x.inject(:+) }
end

text str: 'Title', font: 'Sans 64'  # who knows where this extends to - don't care and it will change
text str: 'subtitle', y: plus(last_y, last_height) + 50 #50 pixels below where the subtitle ends
# last_y is now going to be set to subtitle's y
```

As a starter, we can expose "extents" that gets set after each `text` method. That makes the text syntax a bit better.

**Comment by sparr** [comment]
Wasn't this implemented in v0.0.6 / 2014-12-08 exposing the extent of each text call?

**Comment by andymeneely** [comment]
Yes, although I was going to take it much farther. Haven't really put together a list of what I wanted though and I haven't seen a huge need for it in my own prototyping. Anything you want?

**Comment by sparr** [comment]
Pie in the sky request? I'd love for every method to return the whole object it produced. Not just width and height extents, but also x y location, colors, fonts, angles, etc. I've got use cases for a small fraction of those things, but I've got a gut feeling that just returning everything wouldn't be much more work than returning a few specific things.

**Comment by andymeneely** [comment]
@sparr if you want to work on this that would be great. the `text` method is a good start. `extents` might be a good method/variable that gets set. 

shouldn't be too hard... I think

**Comment by sparr** [comment]
cairo exposes path_extents, stroke_extents, and fill_extents for an active path. if we're only going to return one of them, any thoughts on which? or possibly some manipulation of all of them to produce the actual maximum extents and minimum coordinates
